File: token-validation-jaxrs/src/test/java/no/nav/security/token/support/jaxrs/rest/ProtectedClassResource.java
package no.nav.security.token.support.jaxrs.rest;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.core.Response;
import no.nav.security.token.support.core.api.Protected;

@Path("class/protected")
@Protected
public class ProtectedClassResource {

    @GET
    public Response get() {
        return Response.ok().build();
    }

}

File: token-validation-jaxrs/src/test/kotlin/no/nav/security/token/support/jaxrs/rest/ProtectedClassResource.kt
package no.nav.security.token.support.jaxrs.rest

import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.core.Response.*
import no.nav.security.token.support.core.api.Protected

@Path("class/protected")
@Protected
class ProtectedClassResource {

    @GET
    fun get() = ok().build()
}